### PR TITLE
Change background color on focus

### LINKED ISSUE #2236: [Enhancement] Change background color on focus inside accordion, list, action sheet and dropdown
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
The current solution for focus with a blue outline around elements is not suitable for some of our elements:
- List items 
- Accordion items
- Buttons in action sheet 
- Buttons inside an opened dropdown

Instead we want to change the background of these to be similar to active state for the time being.

A small refactor of dropdown is possibly needed here as the current focus-behavior is not exactly as we want it. Right now, when using the arrow keys the elements are selected. They should instead have focus state when browsed with arrow keys, and only be selected (bold text and checkmark) when enter or space is pressed. 

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [stable branch](https://github.com/kirbydesign/designsystem/tree/stable) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by jkaltoft**
## Which issue does this PR close?

This PR closes #2236 

## What is the new behavior?

Applies alternative focus style. Instead of displaying focus ring the background color is changed by adjusting the state layer opacity.

- [x] Action sheet options
- [x] Accordion
- [x] Tabs on desktop view - awaiting https://github.com/kirbydesign/designsystem/pull/2419

Dropdown will be solved by https://github.com/kirbydesign/designsystem/issues/2342

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/core/src/scss/interaction-state/_focus.scss [hunk 1]
```diff
@@ -36,6 +37,30 @@
   }
 }
 
+// -----------------------------------------------------------------------------
+// In some cases a focus ring is not desirable, e.g., when buttons are stacked
+// (see Action Sheet).
+// As an alternative change the state layer opacity - similar to hover.
+// -----------------------------------------------------------------------------
+@mixin apply-focus-visible-without-ring(
```
Comment (RasmusKjeldgaard) [inline libs/core/src/scss/interaction-state/_focus.scss:45]: Could we consider something like apply-focus-visible-layer or -background (even though not entirely technically true) to make it a bit more descriptive?

**Review by RasmusKjeldgaard — COMMENTED** [review]
Awesome work! I just have a non-blocking comment about the mixin name.

File: libs/core/src/scss/interaction-state/_focus.scss (hunk 1 above)
Comment (jkaltoft) [reply libs/core/src/scss/interaction-state/_focus.scss:45]: Absolutely! 🙂

**Review by RasmusKjeldgaard — APPROVED** [review]
🎉

**Pull request merged**